lambdas/preview/index.py

```diff
@@ -59,6 +62,14 @@ def lambda_handler(request):
     url = request.args['url']
     input_type = request.args.get('input')
     compression = request.args.get('compression')
+
+    parsed_url = urlparse(url, allow_fragments=False)
+    if not (parsed_url.scheme == 'https' and parsed_url.netloc.endswith(S3_DOMAIN_SUFFIX) and
+            parsed_url.username is None and parsed_url.password is None):
+        return make_json_response(400, {
+            'title': 'Invalid S3 URL'
```

Comment:
Suggestion: `'Invalid url=. Expected S3 virtual-host URL.`

Reply:
Just curious, why `url=`? Is that some kind of convention? (That doesn't even match the request syntax...)

Reply:
did i read [the code](https://github.com/quiltdata/quilt/blob/56a172ff299406d94cf3dd9b4da59c2d9c5b6850/lambdas/preview/index.py#L62) wrong? i thought the caller does `foo/preview?url=`. anyway we should reiterate the calling syntax in the message so the caller can diagnose.

Reply:
Oops, you're right, I was looking at the wrong thing.